### PR TITLE
Discover+Notify container engine

### PR DESCRIPTION
**Pull request by asafchen-dig**
adding new key named "user.container-engine" which might contain one of:
"docker"
"podman"
"unknown"

### CODE REVIEW

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/posthog/ContainerEngineStartupActivity.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+class ContainerEngineStartupActivity : StartupActivity {
+    private val timeoutInSec = 5L
+    private val logger = Logger.getInstance(ContainerEngineStartupActivity::class.java)
+    private val isWindows: Boolean = System.getProperty("os.name").startsWith("windows", true)
+
+    enum class ContainerEngine {
+        DOCKER,
+        PODMAN
+    }
```
Comment (arik-dig) [inline ide-common/src/main/kotlin/org/digma/intellij/plugin/posthog/ContainerEngineStartupActivity.kt:17]:
```suggestion
    enum class ContainerEngine {
        DOCKER,
        PODMAN,
        UNKNOWN, // value when cannot determine if have such
        NONE, // value when could determine that have no container engine
    }
```

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/posthog/ActivityMonitor.kt [hunk 2]
```diff
@@ -229,6 +226,13 @@ class ActivityMonitor(private val project: Project) /*: Runnable, Disposable*/ {
         )
     }
 
+    fun registerContainerEngine(containerPlatform: String?) {
```
Comment (arik-dig) [inline ide-common/src/main/kotlin/org/digma/intellij/plugin/posthog/ActivityMonitor.kt:229]: please avoid the nulls.
use within the enum of `ContainerEngine` the NONE and/or UNKOWN

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/posthog/ContainerEngineStartupActivity.kt [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    override fun runActivity(project: Project) {
+        val containerPlatform = getInstalled(project)?.toString()
+        ActivityMonitor.getInstance(project).registerContainerEngine(containerPlatform)
+    }
+
+    private fun getInstalled(project: Project): ContainerEngine? {
```
Comment (arik-dig) [inline ide-common/src/main/kotlin/org/digma/intellij/plugin/posthog/ContainerEngineStartupActivity.kt:24]: please void the null.
```suggestion
    private fun getInstalled(project: Project): ContainerEngine {
```

**Review by arik-dig — APPROVED** [review]
please avoid the nulls, and use dedicated entry within the enum

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/posthog/ContainerEngineStartupActivity.kt (hunk 1 above)
Comment (asafchen-dig) [reply ide-common/src/main/kotlin/org/digma/intellij/plugin/posthog/ContainerEngineStartupActivity.kt]: Added UNKNOWN

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/posthog/ActivityMonitor.kt (hunk 2 above)
Comment (asafchen-dig) [reply ide-common/src/main/kotlin/org/digma/intellij/plugin/posthog/ActivityMonitor.kt]: removed "?"

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/posthog/ContainerEngineStartupActivity.kt (hunk 3 above)
Comment (asafchen-dig) [reply ide-common/src/main/kotlin/org/digma/intellij/plugin/posthog/ContainerEngineStartupActivity.kt]: removed "?"

**Pull request merged**